proof/dleq.go

```diff
@@ -0,0 +1,108 @@
+package proof
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/hash"
+	"github.com/dedis/crypto/random"
+)
+
+// This package provides functionality to create and verify non-interactive
+// zero-knowledge (NIZK) proofs for the equality (EQ) of discrete logarithms (DL).
+// This means, for two values xG and xH one can check that log_{G}(xG) == log_{H}(xH)
+// without revealing the secret value x.
+
+var errorDifferentLengths = errors.New("inputs of different lengths")
+
+// DLEQProof represents a NIZK dlog-equality proof.
+type DLEQProof struct {
+	C  abstract.Scalar // challenge
+	R  abstract.Scalar // response
+	VG abstract.Point  // public commitment with respect to base point G
+	VH abstract.Point  // public commitment with respect to base point H
+}
+
+// NewDLEQProof computes a new NIZK dlog-equality proof by randomly selecting a
```

Comment:
doc: if you want to make it really nice, remember that godoc is very simple. So it could be written like this:

```
// NewDLEQProof computes a new NIZK dlog-equality proof by randomly selecting a
// commitment v, determining the challenge c and the response r:
//   c = H(xG,xH,vG,vH)
//   r = v - cx. 
// It also returns the encrypted base points xG and xH.
```